src/praisonai-agents/praisonaiagents/agents/autoagents.py

```
@@ -238,33 +239,74 @@
-            except ValueError as e:
-                # AutoAgents requires OpenAI for structured output generation
-                raise ValueError(
-                    "AutoAgents requires OpenAI API for automatic agent generation. "
-                    "Please set OPENAI_API_KEY environment variable or use PraisonAIAgents class directly "
-                    "with manually configured agents for non-OpenAI providers."
-                ) from e
+                # Check if we have OpenAI API and the model supports structured output
+                if self.llm and (self.llm.startswith('gpt-') or self.llm.startswith('o1-') or self.llm.startswith('o3-')):
+                    client = get_openai_client()
+                    use_openai_structured = True
+            except:
```

Ruff (0.11.9)

Do not use bare `except`

(E722)